business/apps.go

```diff
@@ -63,14 +125,63 @@ func (in *AppService) GetAppList(namespace string) (models.AppList, error) {
 			Name:         keyApp,
 			IstioSidecar: true,
 		}
-		labels := make(map[string][]string)
+		applabels := make(map[string][]string)
+		svcReferences := make([]*models.IstioValidationKey, 0)
 		for _, srv := range valueApp.Services {
-			joinMap(labels, srv.Labels)
+			joinMap(applabels, srv.Labels)
+			if linkIstioResources {
+				svcVirtualServices := kubernetes.FilterVirtualServices(*linkedResources[kubernetes.VirtualServices], srv.Namespace, srv.Name)
+				svcDestinationRules := kubernetes.FilterDestinationRules(*linkedResources[kubernetes.DestinationRules], srv.Namespace, srv.Name)
+				allFiltered := append(svcVirtualServices, svcDestinationRules...)
+				for _, a := range allFiltered {
+					ref := models.IstioValidationKey{
+						Name:       a.GetObjectMeta().Name,
+						Namespace:  a.GetObjectMeta().Namespace,
+						ObjectType: a.GetTypeMeta().Kind,
+					}
+					svcReferences = append(svcReferences, &ref)
+				}
+			}
+
+		}
+
+		wkdResources := []string{
+			kubernetes.Gateways,
+			kubernetes.AuthorizationPolicies,
+			kubernetes.PeerAuthentications,
+			kubernetes.Sidecars,
+			kubernetes.RequestAuthentications,
+			kubernetes.EnvoyFilters,
 		}
+		wkdReferences := make([]*models.IstioValidationKey, 0)
 		for _, wrk := range valueApp.Workloads {
-			joinMap(labels, wrk.Labels)
+			joinMap(applabels, wrk.Labels)
+			if linkIstioResources {
+				wSelector := labels.Set(wrk.Labels).AsSelector().String()
+				for _, wkdRsc := range wkdResources {
+					filtered := kubernetes.FilterIstioObjectsForWorkloadSelector(wSelector, *linkedResources[wkdRsc])
+					for _, a := range filtered {
+						ref := models.IstioValidationKey{
+							Name:       a.GetObjectMeta().Name,
+							Namespace:  a.GetObjectMeta().Namespace,
+							ObjectType: a.GetTypeMeta().Kind,
+						}
+						exist := false
+						for _, r := range wkdReferences {
+							if r.Name == ref.Name && r.Namespace == ref.Namespace && r.ObjectType == ref.ObjectType {
+								exist = true
+							}
+						}
+						if !exist {
+							wkdReferences = append(wkdReferences, &ref)
+						}
```

Comment:
Are there going to be duplicates here? `wkdReferences` is filled each time with a set of one single type of objects (i.e. Gateways, Sidecars...) and it is impossible to have two equal instances between each set because they have different object type. And within the same set of objects, it is no possible because k8s don't allow that.

Anyway, if that is the case, I'd suggest to use the native `== operator` for comparing references. Golang implements the `DeepEquals()`. So you could do something like this:

```go
for _, r := range wkdReferences {
  exist = exist || *r == ref
}
```

Reply:
Yes, see that we are at application level which may iterate over the workloads.
So, for example, an AuthorizationPolicy that has a selector like "app=travels" would bring same object for multiple workloads like "travels-v1", "travels-v2", "travels-v3".

Perhaps at service and workload level is not necessary, as we don't iterate, but yes, for App is necessary.